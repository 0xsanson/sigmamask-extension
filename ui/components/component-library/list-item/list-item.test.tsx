import * as React from 'react';
import { render } from '@testing-library/react';
import { ListItem } from '.';

describe('ListItem', () => {
  it('should render the ListItem without crashing', () => {
    const { container } = render(<ListItem>Test item</ListItem>);
    expect(container).toMatchSnapshot();
  });

  it('should render the ListItem with additional className', () => {
    const { getByTestId } = render(
      <ListItem data-testid="classname" className="mm-test">
        Test item
      </ListItem>,
    );
    expect(getByTestId('classname')).toHaveClass('mm-list-item mm-test');
  });

  it('should render the ListItem as disabled', () => {
    const { getByTestId } = render(
      <ListItem data-testid="disabled" className="mm-test" isDisabled>
        Test item
      </ListItem>,
    );
    expect(getByTestId('disabled')).toHaveClass('mm-list-item--disabled');
  });

  it('should render the ListItem children', () => {
    const { getByText } = render(
      <ListItem data-testid="children">Test item children</ListItem>,
    );
    expect(getByText('Test item children')).toBeTruthy();
  });
});
