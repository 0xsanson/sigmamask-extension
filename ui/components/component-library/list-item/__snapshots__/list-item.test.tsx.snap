// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render the ListItem without crashing 1`] = `
<div>
  <div
    class="mm-box mm-list-item mm-box--padding-4 mm-box--display-block"
  >
    Test item
  </div>
</div>
`;
